packages/blade/src/components/Checkbox/CheckboxIcon/CheckboxIconWrapperStyles.ts

```diff
@@ -0,0 +1,83 @@
+import type { CSSObject } from 'styled-components';
+import type { CheckboxIconProps } from './CheckboxIcon';
+import type { Theme } from '~components/BladeProvider';
+import { getIn, makeSize, makeSpace } from '~utils';
+
+const variants = {
+  default: {
+    border: {
+      checked: 'colors.brand.primary.500',
+      unchecked: 'colors.brand.gray.400',
+    },
+    background: {
+      checked: 'colors.brand.primary.500',
+      unchecked: 'colors.brand.gray.200',
+    },
+  },
+  disabled: {
+    border: {
+      checked: 'colors.brand.gray.300',
+      unchecked: 'colors.brand.gray.300',
+    },
+    background: {
+      checked: 'colors.brand.gray.300',
+      unchecked: 'colors.brand.gray.300',
+    },
+  },
+  negative: {
+    border: {
+      checked: 'colors.feedback.border.negative.highContrast',
+      unchecked: 'colors.feedback.border.negative.highContrast',
+    },
+    background: {
+      checked: 'colors.feedback.background.negative.highContrast',
+      unchecked: 'colors.feedback.background.negative.lowContrast',
+    },
+  },
+};
+
+type WithChecked = {
+  isChecked: boolean;
+};
+
+export type CheckboxRectProps = Omit<CheckboxIconProps, 'state'> & WithChecked;
```

Comment:
why not inline `WithChecked`? one less indirection, as it's not being used anywhere else

Reply:
Done.